File: 2026_GFG_POTD/FEBRUARY/17-02-2026/Maximum number of overlapping Intervals.py
class Solution:
    def overlapInt(self, arr):
        n = len(arr)
        
        start = sorted([interval[0] for interval in arr])
        end = sorted([interval[1] for interval in arr])
        
        i = 0
        j = 0
        curr_overlap = 0
        max_overlap = 0
        
        while i < n and j < n:
            if start[i] <= end[j]:
                curr_overlap += 1
                max_overlap = max(max_overlap, curr_overlap)
                i += 1
            else:
                curr_overlap -= 1
                j += 1
                
        return max_overlap

File: 2026_GFG_POTD/FEBRUARY/17-02-2026/Maximum number of overlapping Intervals.java
import java.util.*;

class Solution {
    public static int overlapInt(int[][] arr) {

        int n = arr.length;

        int[] start = new int[n];
        int[] end = new int[n];

        for (int i = 0; i < n; i++) {
            start[i] = arr[i][0];
            end[i] = arr[i][1];
        }

        Arrays.sort(start);
        Arrays.sort(end);

        int i = 0, j = 0;
        int current = 0;
        int maxOverlap = 0;

        while (i < n && j < n) {

            if (start[i] <= end[j]) {
                current++;
                maxOverlap = Math.max(maxOverlap, current);
                i++;
            } else {
                current--;
                j++;
            }
        }

        return maxOverlap;
    }
}